install/0000_90_machine-api-operator_04_alertrules.yaml

```diff
@@ -18,25 +18,23 @@ spec:
             severity: critical
           annotations:
             message: "machine {{ $labels.name }} does not have valid node reference"
-
----
-apiVersion: monitoring.coreos.com/v1
-kind: PrometheusRule
-metadata:
-  labels:
-    prometheus: k8s
-    role: alert-rules
-  name: machineapioperator-rules
-  namespace: openshift-machine-api
-spec:
-  groups:
-    - name: general.rules
+    - name: machine-api-operator-metrics-collector-up
+      rules:
+        - alert: MachineAPIOperatorMetricsCollectionFailing
+          expr: |
+             mapi_machine-api-operator_collector_up == 0
```

Comment:
Can we / should we consistently use _ (underscore) here?

Reply:
Yes you should, `-` is an invalid character in prometheus data model. 

> The metric name specifies the general feature of a system that is measured [...] It may contain ASCII letters and digits, as well as underscores and colons. It must match the regex [a-zA-Z_:][a-zA-Z0-9_:]*


Source: https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels